crates/iota-indexer/src/handlers/mod.rs

```diff
@@ -45,3 +53,116 @@ pub struct EpochToCommit {
     pub new_epoch: IndexedEpochInfo,
     pub network_total_transactions: u64,
 }
+
+pub struct CommonHandler<T> {
+    handler: Box<dyn Handler<T>>,
+}
+
+impl<T> CommonHandler<T> {
+    pub fn new(handler: Box<dyn Handler<T>>) -> Self {
+        Self { handler }
+    }
+
+    async fn start_transform_and_load(
+        &self,
+        cp_receiver: iota_metrics::metered_channel::Receiver<(u64, T)>,
+        cancel: CancellationToken,
+    ) -> IndexerResult<()> {
+        let checkpoint_commit_batch_size = std::env::var("CHECKPOINT_COMMIT_BATCH_SIZE")
+            .unwrap_or(CHECKPOINT_COMMIT_BATCH_SIZE.to_string())
+            .parse::<usize>()
+            .unwrap();
```

Comment:
Is the explicit `unwrap()` intended? 

if this fails, then in this task will fail silently without any errors in the `start_objects_snapshot_handler` function.

```rust
spawn_monitored_task!(common_handler.start_transform_and_load(receiver, cancel));
```

```suggestion
            .ok()
            .and_then(|val| val.parse().ok())
            .unwrap_or(CHECKPOINT_COMMIT_BATCH_SIZE);
```